elide-async/src/main/java/com/yahoo/elide/async/service/DBUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.async.service;
+
+import static com.yahoo.elide.core.EntityDictionary.NO_VERSION;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.jsonapi.models.JsonApiDocument;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * Utility class which implements a static method executeInTransaction.
+ */
+@Slf4j
+public class DBUtil {
+
+    /**
+     * This method creates a transaction from the datastore, performs the DB action using
+     * a generic functional interface and closes the transaction.
+     * @param dataStore Elide datastore retrieved from Elide object
+     * @param action Functional interface to perform DB action
+     * @return Object Returns Entity Object (AsyncQueryResult or AsyncResult)
+     */
+    protected static Object executeInTransaction(Elide elide, DataStore dataStore, Transactional action) {
```

Comment:
If you only need the Settings, don't pass through Elide object itself.  Change the signature (and all the related plumbing to pass in ElideSettings instead of Elide).